### DISCUSSION TITLE
Create simple PAGTextLayer, bounds are (0, 0, 0, 0)

### DISCUSSION THREAD

**Discussion by bojanstef** [Tencent/libpag#1703] [Q&A] [answered]
I have a very simple UIViewController and I'm trying to present text like the follow code snippet. Yet the print statements output zero bounds. Is it necessary to load a file? Shouldn't I be able to generate a PAGComposition with just a PAGTextLayer?
```
composition bounds: (0.0, 0.0, 0.0, 0.0)
textLayer bounds: (0.0, 0.0, 0.0, 0.0)
```

```swift
import UIKit
import libpag

class ViewController: UIViewController {
  override func viewDidLoad() {
    super.viewDidLoad()
    // Do any additional setup after loading the view.
    view.backgroundColor = .systemBackground

    // Setup PAGView
    let pagView = PAGView()
    pagView.backgroundColor = .systemGray6
    pagView.frame = CGRect(x: 0, y: 0, width: view.bounds.width, height: 256)
    pagView.center = view.center

    // Setup PAGTextLayer
    let textLayer = PAGTextLayer()
    let font = PAGFont()
    font.fontFamily = "ArialMT"
    font.fontStyle = "Regular"
    textLayer.setFont(font)
    textLayer.setFill(.blue)
    textLayer.setStroke(.white)
    textLayer.setText("My name is Bojan.")
    textLayer.setFontSize(32)

    // Setup PAGComposition
    let composition = PAGComposition()
    composition.setContentSize(CGSize(width: view.bounds.width, height: 256))

    // Add PAGTextView -> PAGComposition -> PAGView -> UIView
    composition.add(textLayer)
    pagView.setComposition(composition)
    view.addSubview(pagView)

    print("composition bounds:", composition.getBounds())
    print("textLayer bounds:", textLayer.getBounds())
  }
}

```

Xcode: Version 14.3 (14E222b)
iOS: 16.6 (20G75)
libpag: 4.3.21
swift-driver version: 1.75.2 Apple Swift version 5.8 (swiftlang-5.8.0.124.2 clang-1403.0.22.11.100)
Target: arm64-apple-macosx13.0

**Comment by kevingpqi123** [comment]
PAGTextLayer does not support external creation and needs to be added by the designer in After Effects.

**Comment by domchen** [comment, marked as answer]
PAG is primarily designed for rendering animations created in After Effects. It is not recommended to create animations through coding. The PAGComposition APIs enable the combination of multiple PAG files into a single entity, with the primary objective of facilitating seamless animation integration rather than creating animations from scratch. So you can not create a PAGTextLayer.

**Reply by bojanstef** [reply]
Acknowledged. Thank you.
